### ISSUE TITLE
Suggestion: Luis Entity Attribute

### ISSUE THREAD

**Issue by warrenbuckley** [Microsoft/BotBuilder#24]
Hiya
After waking up this morning. I had an idea or suggestion that to help with mapping Luis Entity Items was to decorate properties with an attribute to map the Entity type to.

```csharp
[LuisEntity("User")]
property string User { get; set; }

[LuisEntity("builtin.datetime.time")]
property TimeSpan TimeToConvert { get; set; }
```

**Comment by Stevenic** [comment]
Interesting... I'm the Node.js guy here so take my thoughts with a grain of salt :) At first blush I like that idea but I do wonder a bit about how that would ultimately feel from a developer perspective... From a LUIS perspective your model maps directly to a C# class. That means that each method is potentially a handler for a separate LUIS intent (or now LUIS action.) With this proposal you'd have to say [set property, set property, call method] anytime an intent was triggered. That feels a little weird. My originaly proposal to the team was that they model Luis Dialogs after .Net style Web Controlers and part of that proposal was that they do what Web Controllers do and map entities to arguments on the handler they invoke. I still feel like this is ultimately the cleaner approach. Especially, in light of the new LUIS Actions feature.

**Comment by warrenbuckley** [comment]
No problem @Stevenic if you feel that this is cleaner then thats fine. Feel free to close the issue. I was just spitballing idea/s after waking up after my last nights hacking for a way to map entity items, but its not major deal to map it manually myself, was just thinking of some extra sugar/nice-ness.

**Comment by Stevenic** [comment]
We can leave it open and let the C# guys weigh in. Like I said I'm the Node guy. :)

**Comment by chrimc62** [comment]
That is a good suggestion.  We will look into adding this in the next version.

**Comment by Stevenic** [comment]
Moved to UserVoice:

https://feedback.botframework.com/forums/354222-bot-builder-sdk/suggestions/13463430--c-suggestion-luis-entity-attribute

**Comment by ArtriciaDM** [comment]
Thanks

Artricia Harris
Closed #24.

---
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/Microsoft/BotBuilder/issues/24#event-630690022
